pymodm/base/options.py

```diff
@@ -76,8 +76,8 @@ def auto_dereference(self):
     def auto_dereference(self, auto_dereference):
         """Turn automatic dereferencing on or off."""
         for field in self.get_fields():
-            if isinstance(field, (EmbeddedDocumentField,
-                                  EmbeddedDocumentListField)):
+            if isinstance(field, (EmbeddedModelField,
+                                  EmbeddedModelListField)):
```

Comment:
I believe this should just be `if isinstance(field, RelatedEmbeddedModelFieldsBase):`. That way it will work with both the new and old classes.